### PR TITLE
CDAP-4566 separate startup checks and run by init scripts

### PR DESCRIPTION
**Pull request by albertshau**
Moving startup checks from MasterServiceMain to its own class so
that they are run in a separate command. This is so we can call
it separately in the init script. This will make it so that a
startup failure will cause the init scripts to exit with a failure,
rather than having the scripts exist with a success, then have the
master fail soon after.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-RBT577

**Comment by jwang47** [comment]
LGTM

File: cdap-common/bin/service [hunk 1]
```diff
@@ -103,6 +103,10 @@ _start_java() {
     cdap_set_hbase || exit 1
     # Master requires this local directory
     cdap_check_or_create_master_local_dir || die "Could not create Master local directory"
+    "${JAVA}" "${JAVA_HEAPMAX}" -Dhive.classpath=${HIVE_CLASSPATH} ${OPTS} -cp ${CLASSPATH} co.cask.cdap.master.startup.MasterStartupTool </dev/null >>${loglog} 2>&1
```
Comment (dereklwood) [inline cdap-common/bin/service]: ``HIVE_CLASSPATH`` or ``EXPLORE_CLASSPATH`` ?  and do you not need ``-Dexplore.conf.files`` ?

Comment (albertshau) [inline cdap-common/bin/service]: oops good catch.  Will be sure to test this

**Pull request merged**